### PR TITLE
Refactor syscall handing

### PR DESCRIPTION
**Pull request by fyrchik**
Depends on #1711 .

1. Introduce `neogointernal` package with `Syscall*` functions.
2. Process syscall wrappers as usual Go functions.
3. For native contract wrappers inlinining is done twice (for the func itself + inner syscall).

### CODE REVIEW

File: pkg/compiler/codegen.go [hunk 1]
```diff
@@ -1694,6 +1694,9 @@ func transformArgs(fun ast.Expr, args []ast.Expr) []ast.Expr {
 		if f.Sel.Name == "FromAddress" {
 			return args[1:]
 		}
+		if strings.HasPrefix(f.Sel.Name, "Syscall") {
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:1697]: Won't this affect user-defined `Syscall*` functions?

File: pkg/compiler/codegen.go [hunk 2]
```diff
@@ -1523,12 +1523,12 @@ func (c *codegen) getByteArray(expr ast.Expr) []byte {
 	}
 }
 
-func (c *codegen) convertSyscall(expr *ast.CallExpr, api, name string) {
-	syscall, ok := syscalls[api][name]
-	if !ok {
-		c.prog.Err = fmt.Errorf("unknown VM syscall api: %s.%s", api, name)
-		return
+func (c *codegen) convertSyscall(expr *ast.CallExpr) {
+	for i := len(expr.Args) - 1; i > 0; i-- {
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:1527]: So in `f(a, b ,c)` we're calculating `c` first? It can break things.

File: pkg/compiler/func_scope.go [hunk 3]
```diff
@@ -139,6 +144,10 @@ func (c *codegen) countLocalsInline(decl *ast.FuncDecl, pkg *types.Package, f *f
 						size++
 					}
 				}
+				// Variadic with direct var args.
+				if sig.Variadic() && !n.Ellipsis.IsValid() {
```
Comment (roman-khimov) [inline pkg/compiler/func_scope.go:148]: Different problem?

**Comment by roman-khimov** [comment]
And it seems to be fixing #941, but it's not mentioned anywhere.

File: pkg/compiler/codegen.go (hunk 1 above)
Comment (fyrchik) [reply pkg/compiler/codegen.go:1697]: Fixed.

File: pkg/compiler/codegen.go (hunk 2 above)
Comment (fyrchik) [reply pkg/compiler/codegen.go:1527]: Not if we use it as we are currently, but fixed.

File: pkg/compiler/func_scope.go (hunk 3 above)
Comment (fyrchik) [reply pkg/compiler/func_scope.go:148]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**